server/app.js

```diff
@@ -1,15 +1,152 @@
+require("dotenv").config();
 const express = require("express");
+const cors = require("cors");
+const mongoose = require("mongoose");
+const Students = require("./models/students");
+const Course = require("./models/course");
 
 const app = express();
 
+app.use(cors());
 app.use(express.json());
+app.use((req, res, next) => {
+  const { method, url } = req;
+  console.log("Middleware running now!");
+  console.log(`${method} for the path "${url}" of db "exercise"`);
+  next();
+});
+
+app.get("/course", (req, res) => {
+  Course.find()
+    .then((course) => {
+      res.json(course);
+      res.status(200);
+      console.log(course);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({ error: `Internal Server Error ${error}` });
+    });
+});
+
+app.get("/course/:courseId", (req, res) => {
+  const { courseId } = req.params;
+
+  Course.findById(courseId)
+    .then((post) => {
+      res.status(200);
+      res.json(post);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server error ${error}`,
+      });
+    });
+});
+
+app.get("/students", (req, res) => {
+  Students.find()
+    .then((course) => {
+      res.status(200);
+      res.json(course);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json(error);
+    });
+});
+
+app.get("/students/:studentsId", (req, res) => {
+  const { studentsId } = req.params;
+  Students.findById(studentsId)
+    .then((post) => {
+      res.status(200);
+      res.json(post);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server error ${error}`,
+      });
+    });
+});
+
+app.get("/course/:id/students/:id", (req, res) => {
```

Comment:
This will be problematic because you assign the name `id` to two different parameters. You already have the same functionality on the endpoint starting in line 32.
I assume this has to do with the weird description of the task. My bad, sorry. You can already retrieve a specific student even without the `courseId`. The idea behind that task was to make you look for a course _and_ a student. For that you'd have to have two different parameters (e.g. 'courseId` and `studentId`) and then look up two different entries in your database. I realised too late, that it won't make much sense in the exercise